orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -161,18 +152,33 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
     return stages
   }
 
-  private Map makeIncrementalCapacity(Map originalCapacity, Integer p) {
+  static ResizeStrategy.Source getSource(TargetServerGroupResolver targetServerGroupResolver,
+                                         RollingRedBlackStageData stageData,
+                                         Map baseContext) {
+    if (stageData.source) {
+      return new ResizeStrategy.Source(
+        region: stageData.source.region,
+        serverGroupName: stageData.source.serverGroupName ?: stageData.source.asgName,
+        credentials: stageData.credentials ?: stageData.account,
+        cloudProvider: stageData.cloudProvider
+      )
+    }
+
+    // no source server group specified, lookup current at build time
```

Comment:
I don't follow the build time comment - do you mean it's build time because we're inserting a dynamic stage now?

Reply:
I need to actually rejig this further ... and move the source determination to something along the lines of ancestor + `DetermineTargetServerGroup` and make it determined at runtime when the deploy stage starts.

It's build time right now because this is all happening when the pipeline / stages are constructed.

Reply:
Ah sounds good

Reply:
Turns out v3 makes this a-ok ... the determination isn't made until the deploy stage is constructed.

I'm otherwise happy with this PR, alright to merge? 

We've had a request to do something other than a wait between rollouts, so I'll start looking at supporting pipelines next.